packages/app/src/pages/Debug.vue

```diff
@@ -27,12 +29,25 @@ subscription Debug_specsChange {
 `
 
 gql `
-query Debug {
+query Debug($runNumber: Int!, $nextRunNumber: Int!, $hasNextRun: Boolean!) {
   ...DebugSpecs
 }
 `
 
-const query = useQuery({ query: DebugDocument })
+const relevantRuns = useRelevantRun()
+
+const variables = ref({ runNumber: -1, nextRunNumber: -1, hasNextRun: false })
+
+const query = useQuery({ query: DebugDocument, variables, pause: true })
+
+watchEffect(() => {
+  if (relevantRuns.value.current) {
+    variables.value.runNumber = relevantRuns.value.current
+    variables.value.hasNextRun = !!relevantRuns.value.next
+    variables.value.nextRunNumber = relevantRuns.value.next || -1
+    query.executeQuery()
```

Comment:
It might be a bit more intuitive to have `variables` be a computed value, and watch only `variables.runNumber` to trigger `query.executeQuery()` when needed?